### PR TITLE
check_jobs return list by default

### PR DESCRIPTION
**Pull request by Tianhao-Gu**
make check_jobs return a sorted list by default
https://github.com/kbase/execution_engine2/issues/80

### CODE REVIEW

File: execution_engine2.spec [hunk 1]
```diff
@@ -314,29 +314,37 @@
 
         /*
             job_states - states of jobs
+
+            could be mapping<job_id, JobState> or list<JobState>
         */
         typedef structure {
-            mapping<job_id, JobState> job_states;
+            UnspecifiedObject job_states;
```
Comment (bio-boris) [inline execution_engine2.spec]: Shouldn't list be a  list<JobStates> ?

Comment (bio-boris) [reply execution_engine2.spec]: Oh I see.. It can return both...

Comment (bio-boris) [reply execution_engine2.spec]: But then this breaks what @briehl said in https://github.com/kbase/execution_engine2/issues/49

File: lib/execution_engine2/SDKMethodRunner.py [hunk 2]
```diff
@@ -835,9 +835,14 @@ def check_jobs(self, job_ids, check_permission=True, projection=None):
 
             job_states[str(job.id)] = mongo_rec
 
+        job_states = dict(sorted(job_states.items()))
+
+        if return_list:
+            job_states = list(job_states.values())
```
Comment (bio-boris) [inline lib/execution_engine2/SDKMethodRunner.py]: What would happen if the string "False" was passed in here?

File: lib/execution_engine2/SDKMethodRunner.py [hunk 3]
```diff
@@ -835,9 +835,14 @@ def check_jobs(self, job_ids, check_permission=True, projection=None):
 
             job_states[str(job.id)] = mongo_rec
 
+        job_states = dict(sorted(job_states.items()))
```
Comment (bio-boris) [inline lib/execution_engine2/SDKMethodRunner.py]: Does this return in the order that the jobs  were requested? For example

1000,500,300 would return the records as 1000,500,300 and not as 300,500,1000

File: lib/execution_engine2/SDKMethodRunner.py (hunk 2 above)
Comment (Tianhao-Gu) [reply lib/execution_engine2/SDKMethodRunner.py]: 👍 hopefully, no one does that. I found your `parse_bool_from_string`. 💯

File: lib/execution_engine2/SDKMethodRunner.py (hunk 3 above)
Comment (Tianhao-Gu) [reply lib/execution_engine2/SDKMethodRunner.py]: 💯 added sort_by to `get_jobs`

File: execution_engine2.spec (hunk 1 above)
Comment (Tianhao-Gu) [reply execution_engine2.spec]: yea. I updated it as `list` since it's the default return type now.

**Comment by Tianhao-Gu** [comment]
@bio-boris Could you take another look?

File: lib/execution_engine2/SDKMethodRunner.py [hunk 4]
```diff
@@ -838,9 +838,12 @@ def check_jobs(self, job_ids, check_permission=True, projection=None):
 
             job_states[str(job.id)] = mongo_rec
 
+        if return_list is not None and SDKMethodRunner.parse_bool_from_string(return_list):
+            job_states = list(job_states.values())
```
Comment (bio-boris) [inline lib/execution_engine2/SDKMethodRunner.py]: I think you need to iterate over the job ids, like 100,900,200,9999 and return them in that order like
list = [{job_state_for_id_100}, {job_state_for_id_900}, {'maybe this job state is missing'},{job_state_for_id_999}]

File: lib/execution_engine2/SDKMethodRunner.py (hunk 2 above)
Comment (bio-boris) [reply lib/execution_engine2/SDKMethodRunner.py]: Ah did you use it? I swear I saw you use it in a diff commit and now its gone from this PR or maybe its just too late at night right now.

File: lib/execution_engine2/SDKMethodRunner.py (hunk 4 above)
Comment (Tianhao-Gu) [reply lib/execution_engine2/SDKMethodRunner.py]: @bio-boris I added a `order_by` in `get_jobs`. So the default jobs returned are sorted by id.

File: lib/execution_engine2/SDKMethodRunner.py (hunk 2 above)
Comment (Tianhao-Gu) [reply lib/execution_engine2/SDKMethodRunner.py]: @bio-boris hmmm no it's there. `if return_list is not None and SDKMethodRunner.parse_bool_from_string(return_list):`

**Comment by Tianhao-Gu** [comment]
@bio-boris check_jobs should now return jobs in same order as the input.

**Pull request merged**